src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+enum Shape {
+  'Triangle' = 'triangle',
+  'Circle' = 'circle',
+  'Rectangle' = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sorted: number[] = [a, b, c].sort((x: number, y: number) => x - y);
 
-export class Circle implements Figure {}
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign them to the class properties inside the constructor.